### ISSUE TITLE
CreateIndex can't resolve MetastoreRelation

### ISSUE THREAD

**Issue by linhong-intel** [Intel-bigdata/OAP#485]
If use HiveCatalog, lookupRelation will return MetastoreRelation.
Now we only can resolve SimpleCatalogRelation.

Maybe below code can fix this:
```
   val relation =
      EliminateSubqueryAliases(sparkSession.sessionState.catalog.lookupRelation(table)) match {
        case r: SimpleCatalogRelation => (new FindDataSourceTable(sparkSession))(r)
        case other => sparkSession.sessionState.catalog match {
          case c: HiveSessionCatalog => c.ParquetConversions(other)
          case _ => other
        }
      }
```

**Comment by adrian-wang** [comment]
I propose shift the resolve and conversion to Spark itself, as other operations do. In this case, Let's get rid of `RunnableCommand`, create corresponding `LogicalPlan` and `PhysicalPlan` for index operations. We only create `LogicalPlan` containing the relation as its child, and Spark will resolve all kinds of relations and do all possible conversion. Then we construct `PhysicalPlan` in `SparkPlanner`.

Assign this to @Moonlit-Sailor

**Comment by adrian-wang** [comment]
closed via #499
